examples/bso_server/mnist.py

```diff
@@ -0,0 +1,273 @@
+from __future__ import print_function
+
+import argparse
+import os
+
+from tensorboardX import SummaryWriter
+from torchvision import datasets, transforms
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from zeus.monitor import ZeusMonitor
+from zeus.optimizer import GlobalPowerLimitOptimizer
+from zeus.optimizer.batch_size.client import BatchSizeOptimizer
+from zeus.optimizer.batch_size.common import JobSpecIn
+
+
+WORLD_SIZE = int(os.environ.get("WORLD_SIZE", 1))
+
+job = {
+    "job_id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
+    "seed": 1,
+    "default_batch_size": 256,
+    "batch_sizes": [32, 64, 256, 512, 1024, 4096, 2048],
+    "eta_knob": 0.5,
+    "beta_knob": 2,
+    "target_metric": 0.5,
+    "high_is_better_metric": True,
+    "max_epochs": 100,
+    "num_pruning_rounds": 2,
+    "window_size": 5,
+    "mab_prior_mean": 0,
+    "mab_prior_precision": 0,
+    "mab_seed": 123456,
+    "mab_num_exploration": 2,
+    "gpu_model": "A100",
+}
+
+monitor = ZeusMonitor(gpu_indices=[torch.cuda.current_device()])
+plo = GlobalPowerLimitOptimizer(monitor)
+bso = BatchSizeOptimizer(
+    monitor=monitor,
+    server_url="http://127.0.0.1:8000",
+    job_in=JobSpecIn.parse_obj(job),
```

Comment:
1. I copied from my test code, I can fix it.
2. I wanted to hide some fields that user doesn't have to care about, such as GPU model, number of GPU, max power. I took it from here (https://fastapi.tiangolo.com/tutorial/response-model/#add-an-output-model)

Reply:
`UserIn` and `UserOut` in the link you gave me can have a strange name because it is still not exposed to the user (i.e., it's server code). In theory it can be called `JaeWonChung` and the user will never know. That's not the case for us here. The user of the Zeus batch size optimizer will need to import and instantiate the job spec class, meaning it's our **public API**. Those should have the cleanest and more intuitive names.